### PR TITLE
Implement toString() for values

### PR DESCRIPTION
**Pull request by gimantha**
## Purpose
This PR introduces toString() function which returns a simple, human-readable representation of the value as a string.

```ballerina
map<string> stringMap = new;
stringMap["firstString"] = "ABC";
string mapAsString = stringMap.toString();
```

## Approach
From toString() external function, stringValue() of RefValue interface is called to generate the string.

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Required Balo version update
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [x] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: langlib/lang.value/src/main/ballerina/lang.value/value.bal [hunk 1]
```diff
@@ -14,6 +14,34 @@
...
-public function toString (any|error value) returns string {
-    return "I am toString";
-}
+# Returns a simple, human-readable representation of `value` as a string.
+# - if `value` is a string, then returns `value`
+# - if `value` is `()`, then returns an empty string
+# - if `value` is boolean, then the string `true` or `false`
+# - if `value` is an int, then return `value` represented as a decimal string
+# - if `value` is a float or decimal, then return `value` represented as a decimal string,
+#   with a decimal point only if necessary, but without any suffix indicating the type of `value`;
```
Comment (anupama-pathirage) [inline langlib/lang.value/src/main/ballerina/lang.value/value.bal]:
```suggestion
#   with a decimal point only if necessary, but without any suffix indicating the type of `value`
```

File: langlib/lang.value/src/main/ballerina/lang.value/value.bal [hunk 2]
```diff
@@ -14,6 +14,34 @@
...
+# - if `value` is table, TBD
+# - if `value` is an error, then a string consisting of the following in order
+#     1. the string `error`
+#     2. a space character
+#     3. the reason string
+#     4. if the detail record is non-empty
+#         1. a space character
+#         2. the result of calling toString on the detail record
+# - if `value` is an object, then
+#     - if `value` provides a toString method with a string return type and no required methods,
```
Comment (anupama-pathirage) [inline langlib/lang.value/src/main/ballerina/lang.value/value.bal]:
```suggestion
#     - if `value` provides a `toString` method with a string return type and no required methods,
```

File: langlib/lang.value/src/main/java/org/ballerinalang/langlib/value/ToString.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+import org.ballerinalang.jvm.values.RefValue;
+import org.ballerinalang.model.types.TypeKind;
+import org.ballerinalang.natives.annotations.Argument;
+import org.ballerinalang.natives.annotations.BallerinaFunction;
+import org.ballerinalang.natives.annotations.ReturnType;
+
+/**
+ * Returns a simple, human-readable representation of the given value as a String.
+ *
+ * @since 0.89
```
Comment (anupama-pathirage) [inline langlib/lang.value/src/main/java/org/ballerinalang/langlib/value/ToString.java]: 1.0?

Comment (gimantha) [reply langlib/lang.value/src/main/java/org/ballerinalang/langlib/value/ToString.java]: fixed

File: bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/values/ErrorValue.java [hunk 4]
```diff
@@ -64,7 +65,7 @@ public ErrorValue(BType type, String reason, Object details) {
 
     @Override
     public String stringValue() {
-        return reason + " " + details.toString();
+        return "error " + reason + " " + Optional.ofNullable(details).map(Object::toString).orElse("");
```
Comment (hasithaa) [inline bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/values/ErrorValue.java]: space after reason is not applicable if the details record is empty.

File: bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/values/MapValueImpl.java [hunk 5]
```diff
@@ -361,7 +361,7 @@ public String stringValue() {
                             continue;
                         }
                         String fieldName = field.getKey();
-                        V fieldVal = get((K) fieldName);
+                        V fieldVal = get(fieldName);
                         sj.add(fieldName + ":" + getStringValue(fieldVal));
```
Comment (hasithaa) [inline bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/values/MapValueImpl.java]: replace : with =

File: bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/values/ErrorValue.java (hunk 4 above)
Comment (gimantha) [reply bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/values/ErrorValue.java]: fixed

File: bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/values/MapValueImpl.java (hunk 5 above)
Comment (gimantha) [reply bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/values/MapValueImpl.java]: Do we need this? now the object are represented with ObjectValues, not by maps

Comment (hasithaa) [reply bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/values/MapValueImpl.java]: @SupunS wdyt ?

Comment (gimantha) [reply bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/values/MapValueImpl.java]: As per the offline chat with @SupunS  we can remove that..

**Pull request merged**